### PR TITLE
Switch to Go 1.16, Remove Statik and switch to Embed

### PR DESCRIPTION
**Pull request by dvaldivia**
Removes the dependency on statik, we still need to copy the resources around sadly, but now we use the native `embed` for embedding static files

### CODE REVIEW

File: kubectl-minio/cmd/resources/common.go [hunk 1]
```diff
@@ -42,12 +41,16 @@ import (
...
+	"embed"
+
 	"github.com/minio/kubectl-minio/cmd/helpers"
-	_ "github.com/minio/kubectl-minio/statik"
 	apiextensionv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	apiextensionv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
 )
 
+//go:embed yamls/*
+var fs embed.FS
```
Comment (dvaldivia) [inline kubectl-minio/cmd/resources/common.go:52]: this is the new special sauce

File: Makefile [hunk 2]
```diff
@@ -70,9 +70,8 @@ regen-crd-docs:
 	@crd-ref-docs --source-path=${GOPATH}/src/github.com/minio/operator/pkg/apis --config=docs/templates/config.yaml --renderer=asciidoctor --output-path=docs/crd.adoc --templates-dir=docs/templates/asciidoctor/
 
 statik:
```
Comment (harshavardhana) [inline Makefile:72]:
```suggestion
embed:
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: resources/assets.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
```
Comment (harshavardhana) [inline resources/assets.go:17]:
```suggestion
package assets
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by bexsoft — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Review by nitisht — APPROVED** [review]
LGTM & Tested

**Pull request merged**